torture.c

```diff
@@ -8,18 +8,72 @@
 #include <stdlib.h>
 #include <time.h>
 #include <unistd.h>
+#include <string.h>
 #include <math.h>
 
+char buffer[1024];
+size_t buffer_pos = 0;
```

Comment:
Can we pull these into `main` please? Did I miss a strong need for global variables here?